### ISSUE TITLE
Custom build strategy ignores tag of DockerImage output

### ISSUE THREAD

**Issue by mmilata** [openshift/origin#3615]
When running custom build with `DockerImage` output, the `$OUTPUT_IMAGE` variable passed to the build container is missing the given docker tag. The tag is missing from `oc describe build` output as well.

The tag was not ignored in OpenShift 0.5.x and furthermore some jsons in examples/ dir refer to DockerImages including tags which indicates that this is a bug.

Encountered on OpenShift 1.0.0 built from source running on RHEL 7.1.

Steps to reproduce (the build image does not actually build anything, just demonstrates that the tag is missing in OUTPUT_IMAGE):
```
# cat Dockerfile 
FROM fedora

CMD ["/bin/bash", "-c", "echo ${OUTPUT_REGISTRY}, ${OUTPUT_IMAGE}"]

# docker build -t buildimage .
...

# cat build.json 
{
  "apiVersion": "v1",
  "kind": "Build",
  "metadata":{
    "name": "my-build"
  },
  "spec": {
    "source": {
      "type": "Git",
      "git": {
        "uri": "https://github.com/mmilata/docker-hello-world.git"
      }
    },
    "output": {
      "to": {
        "kind": "DockerImage",
        "name": "myregistry.org/namespace/name:tag"
      }
    },
    "strategy": {
      "type": "Custom",
      "customStrategy": {
        "from": {
          "kind": "ImageStreamTag",
          "name": "buildimage:latest"
        },
        "exposeDockerSocket": true
      }
    }
  }
}

# oc create -f build.json 
builds/my-build

# oc build-logs my-build
myregistry.org, namespace/name

# oc describe build my-build
...
Output Spec:            myregistry.org/namespace/name
...
```

**Comment by csrwng** [comment]
@bparees can you please assign to me?

**Comment by csrwng** [comment]
@bparees can't reproduce this one anymore. It was likely fixed with the move to using v1 internally.

**Comment by bparees** [comment]
thanks @csrwng, closing.  @mmilata please reopen if this is still an issue for you on current code.

**Comment by mmilata** [comment]
Verified fixed in 1.0.3. Thanks!
